### PR TITLE
Metric - registrations IVR vs USSD

### PR DESCRIPTION
**Pull request by gsvr**

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review. Note I wasn't sure how the sources have been named, so we may need to change that. It may be better to just include the source name in the metric we fire.

File: registrations/models.py [hunk 1]
```diff
@@ -94,6 +94,21 @@ def fire_created_metric(sender, instance, created, **kwargs):
 
 
 @receiver(post_save, sender=Registration)
+def fire_source_metric(sender, instance, created, **kwargs):
+    from .tasks import fire_metric
+    if created and 'ussd' in instance.source.name:
```
Comment (imsickofmaps) [inline registrations/models.py]: please can we user `source.user.username` instead? That's known to be a slug, and should be descriptive too. Then you won't have to have this logic in here.

**Comment by imsickofmaps** [comment]
@gsvr see my comment - it'll affect a few things

**Comment by gsvr** [comment]
@imsickofmaps Check last two commits, please. Wasn't clear if you wanted the metric name to be `source.user.<username>`, I just used `source.<username>`.

File: hellomama_registration/utils.py [hunk 2]
```diff
@@ -206,4 +207,10 @@ def get_available_metrics():
     available_metrics.extend(settings.METRICS_REALTIME)
     available_metrics.extend(settings.METRICS_SCHEDULED)
 
+    users = User.objects.all()
+    for user in users:
+        print(user.username)
```
Comment (imsickofmaps) [inline hellomama_registration/utils.py]: Stray print

File: hellomama_registration/utils.py [hunk 3]
```diff
@@ -206,4 +207,10 @@ def get_available_metrics():
...
+        available_metrics.append(
+            "registrations.source.%s.sum" % user.username)
```
Comment (imsickofmaps) [inline hellomama_registration/utils.py]: username's may have `@`s in them, this may cause problems

Comment (imsickofmaps) [inline hellomama_registration/utils.py]: just do the Source one's - that should avoid the issue

**Comment by bruskiza** [comment]
@smn @imsickofmaps poink WRT this.
I know there's a lot going on at the moment, but this has been outstanding for a while.

**Comment by smn** [comment]
Is it ready for review and should I be doing the reviews? Happy to but just not sure at this point. @gsvr & @imsickofmaps is this still something you're planning on attending & wrapping up?

**Comment by imsickofmaps** [comment]
Back on our radar. Will address next week.

**Comment by gsvr** [comment]
@imsickofmaps Please review again.

**Comment by imsickofmaps** [comment]
merge develop in 👍  once merge conflicts fixed

**Pull request merged**